src/demuxer.rs

```diff
@@ -170,25 +171,149 @@ fn track_entry_codec_id(t: &TrackEntry) -> Option<String> {
         "V_AV1" => Some("av1".to_owned()),
         "V_VP8" => Some("vp8".to_owned()),
         "V_VP9" => Some("vp9".to_owned()),
+        "V_UNCOMPRESSED" => Some("uncompressed".to_owned()),
         _ => None,
     }
 }
 
 fn track_entry_video_kind(t: &TrackEntry) -> Option<MediaKind> {
     // TODO: Validate that a track::video exists for track::type video before.
+
     if let Some(ref video) = t.video {
+        let format = if let Some(ref colour) = video.colour {
+            let chroma = &[
+                Chromaton::new(0, 0, false, 8, 0, 0, 1),
+                Chromaton::new(
+                    colour.cb_subsampling_vert.unwrap_or_else(|| 0) as u8,
+                    colour.cb_subsampling_horz.unwrap_or_else(|| 0) as u8,
+                    false,
+                    0,
+                    8,
+                    0,
+                    1,
+                ),
+                Chromaton::new(
+                    colour.cb_subsampling_vert.unwrap_or_else(|| 0) as u8,
+                    colour.cb_subsampling_horz.unwrap_or_else(|| 0) as u8,
+                    false,
+                    0,
+                    8,
+                    0,
+                    1,
+                ),
+            ];
+
+            let f = Arc::new(Formaton::new(
+                colour_space_to_color_model(video.colour_space.as_ref())
+                    .unwrap_or_else(|| (ColorModel::Trichromatic(TrichromaticEncodingSystem::RGB))),
+                chroma,
+                0,
+                false, // is RGB
+                false,
+                false,
+            ));
+
+            f.set_primaries(
+                FromPrimitive::from_u64(colour.primaries.unwrap_or_else(|| 2)).unwrap(),
+            );
+            f.set_xfer(
+                FromPrimitive::from_u64(colour.transfer_characteristics.unwrap_or_else(|| 2))
+                    .unwrap(),
+            );
+            f.set_matrix(
+                FromPrimitive::from_u64(colour.matrix_coefficients.unwrap_or_else(|| 2)).unwrap(),
+            );
+
+            Some(f)
+        } else {
+            None
+        };
+
         let v = VideoInfo {
             width: video.pixel_width as usize,
             height: video.pixel_height as usize,
-            // TODO parse Colour and/or CodecPrivate to extract the format
-            format: None,
+            format: format,
         };
         Some(MediaKind::Video(v))
     } else {
         None
     }
 }
 
+// parse fourCC value into pixel::ColorModel
+fn colour_space_to_color_model(maybe_cs: Option<&Vec<u8>>) -> Option<ColorModel> {
+    let cs = match maybe_cs {
+        Some(vec) => vec,
+        None => return None,
+    };
+
+    match bytes_to_fourcc_lte(cs[0], cs[1], cs[2], cs[3]) {
```

Comment:
This function has the mapping wrong, and supporting the AVI raw formats would require a lot of additional work.

Let's support generic raw formats first (and force me and @kodabb to write down a draft for it today)